CMakeLists.txt

```
@@ -227,9 +227,15 @@ option(USE_ZMQ "Use ZMQ" OFF)
 option(USE_ZSTD "Use ZSTD" OFF)
 # Ensure that an MKLDNN build is the default for x86 CPUs
 # but optional for AArch64 (dependent on -DUSE_MKLDNN).
-cmake_dependent_option(
-  USE_MKLDNN "Use MKLDNN. Only available on x86, x86_64, and AArch64." ON
-  "CPU_INTEL OR CPU_AARCH64 AND USE_MKLDNN" OFF)
+if(DEFINED USE_MKLDNN)
+  cmake_dependent_option(
+    USE_MKLDNN "Use MKLDNN. Only available on x86, x86_64, and AArch64." ON
+    "CPU_INTEL OR CPU_AARCH64 AND USE_MKLDNN" OFF)
```

this doesn't seem right. if DEFINED USE_MKLDNN should it be:
```suggestion
    "CPU_INTEL AND USE_MKLDNN OR CPU_AARCH64 AND USE_MKLDNN" OFF)
```
otherwise, if `USE_MKLDNN=OFF` it will still be built on x86 right?

---

No, semantic of 3nd argument is whether or not it is user controllable rather whether value is true or false. Also, all boolean opreators have the same precedence in CMake, so original statement was executed as ((CPU_INTEL OR CPU_AARCH64) AND USE_MKLDNN)